pkg/userdata/centos/provider.go

```diff
@@ -168,6 +168,10 @@ write_files:
     systemctl restart systemd-modules-load.service
     sysctl --system
 
+    # Make sure we always disable swap - Otherwise the kubelet won't start
+    systemctl mask swap.target
```

Comment:
Can you mask the swap target to prevent fstab generator creating swap units?
Also you might prevent other units depending on swap.target.
I would propose to remove swap entries from fstab and keep the `swapoff -a` to spare a reboot.

Reply:
To be fair, that is a valid suggestion. However, we already are doing this with Ubuntu (for better or for worse). 

I would propose that we leave this as is, bring centos in line with Ubuntu, and then have a single issue (and eventual PR) to discuss the issues there and align them.

Reply:
I'd argue "its just as subpar elsewhere" is not a very good argument ;) What about instead just updating Ubuntu as well?

Reply:
> argue "its just as subpar elsewhere" is not a very good argument

:-)
I wouldn't disagree. 

> What about instead just updating Ubuntu as well?

I guess we can. I am a little worried about it, but future bugs are part of the game. `sed -i` is your friend...

Reply:
Oh that is going to be a nasty piece of `sed` or `awk`. You need to comment out (but not delete) any lines that contain `swap` in the third column, while not double-commenting out things.

Reply:
Why comment out and not just remove?

Reply:
Two reasons:

1. It isn't really any easier to remove than to comment. You still want to filter only on the 3rd column precisely equal to `swap`.
2. You want to leave it as close to the original as possible, so that if anyone needs to analyze it, you have some form of trail. An SRE can sit them completely confused as to how the template `fstab` is different on this class of instances from every other, and "what in the world changed this?!?". With a comment, it is as close as possible - the original text is there, but commented out - and you can add a comment indicating what commented it.

Reply:
Look at latest updated.

Reply:
`noauto` is an alternative to commenting out.

Reply:
Yes, good point. I think I prefer the explicit comment out though. We aren't trying to say, "don't automatically mount this swap on boot", we are trying to say, "don't ever mount swap on this host."

Reply:
Updated.